packages/generator/src/edmx-to-vdm/common/enum-type.ts

```diff
@@ -18,7 +25,85 @@ export function transformEnumTypesBase(
     return {
       originalName: e.Name,
       typeName,
-      members: e.Member.map(m => m.Name)
+      underlyingType: e.UnderlyingType || 'Edm.Int32',
+      members: parseMember(e)
     };
   });
 }
+
+function parseMember(
+  edmxEnumType: EdmxEnumType
+): Record<string, number | BigNumber> {
+  validateUniqueness(edmxEnumType);
+  validateUnderlyingType(edmxEnumType);
+  edmxEnumType.UnderlyingType = edmxEnumType.UnderlyingType
```

Comment:
This step is done twice and it has nothing to do with parsing the members. I recommend to extract it into `transformEnumTypesBase`:

```ts
// transformEnumTypesBase
...
e.UnderlyingType = e.UnderlyingType || 'Edm.Int32';

return {
  originalName: e.Name,
  typeName,
  underlyingType: e.UnderlyingType,
  members: parseMember(e)
};
```